vsphere/datadog_checks/vsphere/cache.py

```diff
@@ -77,55 +77,74 @@ def set_metadata(self, resource_type, metadata):
 
 class InfrastructureCache(VSphereCache):
     """A VSphere cache dedicated to store the infrastructure data from a user environment.
+    Tags and properties coming from the SOAP API are stored separately for easier manipulation.
+    'tags' is only populated if configured correctly in the conf.yaml file.
     Data is stored like this:
 
     _content = {
-        vim.VirtualMachine: {
-            <MOR_REFERENCE>: <MOR_PROPS_DICT>
+        'mors': {
+            vim.VirtualMachine: {
+                <MOR_REFERENCE>: <MOR_PROPS_DICT>
+            },
+            ...
         },
+        'tags': {
+            <RESOURCE_TYPE>: {
+                <RESOURCE_MOR_ID>: ['<CATEGORY_NAME>:<TAG_NAME>', ...]
+            },
+        }
         ...
     }
     """
 
-    def get_mor_props(self, mor, default=None):
-        # type: (vim.ManagedEntity, Dict[str, Any]) -> Dict[str, Any]
-        mor_type = type(mor)
-        return self._content.get(mor_type, {}).get(mor, default)
-
-    def get_mors(self, resource_type):
-        # type: (Type[vim.ManagedEntity]) -> Iterator[vim.ManagedEntity]
-        return iterkeys(self._content.get(resource_type, {}))
-
-    def set_mor_data(self, mor, mor_data):
-        # type: (vim.ManagedEntity, Dict[str, Any]) -> None
-        mor_type = type(mor)
-        if mor_type not in self._content:
-            self._content[mor_type] = {}
-        self._content[mor_type][mor] = mor_data
-
-
-class TagsCache(VSphereCache):
-    """
-    A VSphere cache dedicated to store the tags data.
-
-    Data is stored like this:
-
-    _content = {
-        <RESOURCE_TYPE>: {
-            <RESOURCE_MOR_ID>: ['<CATEGORY_NAME>:<TAG_NAME>', ...]
-        },
-        ...
-    }
-    """
+    @property
+    def _mors(self):
+        # type: () -> Dict[Type[vim.ManagedEntity], Dict[vim.ManagedEntity, Any]]
+        if 'mors' not in self._content:
+            self._content['mors'] = {}
+        return self._content['mors']
+
+    @_mors.setter
+    def _mors(self, value):
+        # type: (Dict[vim.ManagedEntity, Dict[str, Any]]) -> None
+        self._content['mors'] = value
+
+    @property
+    def _tags(self):
+        # type: () -> ResourceTags
+        if 'tags' not in self._content:
+            self._content['tags'] = {}
+        return self._content['tags']
+
+    @_tags.setter
+    def _tags(self, value):
+        # type: (ResourceTags) -> None
+        self._content['tags'] = value
```

Comment:
I'm not sure there is a need to use `@property` and  `@_tags.setter` instead of directly using `self._content['tags']` and `self._content['mors']`.

Reply:
Yeah that felt cleaner at first but I see your point. I can remove it

Reply:
Actually I need it because `self._content` is cleared before every update. So `self._content['tags']` needs to be created the first time it's accessed.

Getter and Setter allows that in a better way than a defaultdict I think.